src/assets/js/archive.js

```diff
@@ -122,66 +125,98 @@ function createTableCell(content, dataClass) {
 }
 
 function updateTableFailed(os) {
-  const tab = $(`#tab-os-${os}`);
-  tab.find('.loading').text('Failed to load releases. Refresh page to try again.');
+  const tab = document.getElementById(`tab-os-${os}`);
+  if (!tab) return;
+  const loadingElements = tab.querySelectorAll('.loading');
+  loadingElements.forEach(function (element) {
+    element.textContent = 'Failed to load releases. Refresh page to try again.';
+  });
 }
 
 let macOSArm64ArchiveFilename = '';
 
 // Listen for the macOS arm64 download link to be clicked and update
 // the example unzip command with correct arm64 filename
-$('.download-latest-link-macos-arm64').click(function () {
-  // Update inlined filenames in <code> element text nodes with arm64 filename:
-  const fileNamePrefix = 'flutter_';
-  const code = $(`code:contains("${fileNamePrefix}")`);
-  const textNode = $(code).contents().filter(function () {
-    return this.nodeType === 3 && this.textContent.includes(fileNamePrefix);
+(() => {
+  const macDownload = document.querySelector('.download-latest-link-macos-arm64');
+  if (!macDownload) {
+    return;
+  }
+  macDownload.addEventListener('click', function () {
+    // Update inlined filenames in <code> element text nodes with arm64 filename:
+    const textNode = filterTextNodeInCodeElement();
+  
+    textNode.textContent = `unzip ~/Downloads/${macOSArm64ArchiveFilename}`;
   });
+})();
 
-  $(textNode).replaceWith(`unzip ~/Downloads/${macOSArm64ArchiveFilename}`);
-});
+/**
+ * Filters node text that contain a specific prefix.
+ *
+ * @returns {ChildNode} A filtered Node Text.
+ */
+function filterTextNodeInCodeElement() {
+  const codeElements = document.querySelectorAll('code');
+  const filteredElements = Array.from(codeElements).filter(function (element) {
+    return Array.from(element.childNodes).some(function (node) {
+      return node.nodeType === Node.TEXT_NODE && node.textContent.includes(FILE_NAME_PREFIX);
+    });
+  });
+
+  const code = filteredElements[0];
+  const textNode = [...code.childNodes].filter(function (node) {
+    return node.nodeType === Node.TEXT_NODE && node.textContent.includes(FILE_NAME_PREFIX);
+  })[0];
+  return textNode;
+}
```

Comment:
Your new solution looks pretty good and is clever, however it causes a failure if one of the pages with a download button doesn't have a `code` element with the filename, like the [Windows install page](https://docs.flutter.dev/get-started/install/windows/) doesn't. Beyond that, it will only support replacing one instance of the filename. 

How about we take a simpler approach that replaces all instances, and uses a modified regex that supports replacing both the placerholder text and existing filenames. Something like this:

```js
const filenameReplacement = new RegExp(`^(.*?)\\b${FILE_NAME_PREFIX}\\w+_v?[X|0-9]+\\..*`, 'm');

/**
 * Replaces the placeholder text or the old filename in code blocks
 * with the specified {@link archiveFilename}.
 * 
 * @param archiveFilename The new filename to replace the
 *   old one in code blocks with
 */
function replaceFilenameInCodeElements(archiveFilename) {
  const codeElements = document.querySelectorAll('code');

  codeElements.forEach((e) => {
    e.childNodes.forEach((node) => {
      if (node.nodeType === Node.TEXT_NODE &&
          node.textContent.includes(FILE_NAME_PREFIX)) {
        const text = node.textContent;
        node.textContent = text.replace(
            filenameReplacement,
            `$1${archiveFilename}`
        );
      }
    });
  });
}
```

Then we would only need to call this method in the cases that use it.

Reply:
My doubt is where I will call this method. I will make some tests